src/components/App/DisplayPuzzleInstances/index.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import styles from './DisplayPuzzleInstances.module.scss';
+import { PuzzleCustom } from '../../../types/api/puzzles/puzzle';
+import { getAllPuzzleInstances } from '../../../services';
+import { PuzzleInstance } from '@prisma/client';
+import { Button } from '../../Global';
+import { QRGenerator } from '../../App';
+import toast from 'react-hot-toast';
+
+type DisplayPuzzleInstancesProps = {
+  puzzlesList: PuzzleCustom[];
+};
+
+const DisplayPuzzleInstances = ({
+  puzzlesList
+}: DisplayPuzzleInstancesProps) => {
+  const [puzzleId, setPuzzleId] = useState<string>('');
+  const [puzzleInstances, setPuzzleInstances] = useState<PuzzleInstance[]>([]);
+  const [displayPuzzleInstances, setDisplayPuzzleInstances] = useState<
+    PuzzleInstance[]
+  >([]);
+
+  const getPuzzleInstances = async () => {
+    const retrivedPuzzleInstances = await getAllPuzzleInstances();
+    setPuzzleInstances(retrivedPuzzleInstances);
+  };
+
+  useEffect(() => {
+    getPuzzleInstances();
+  }, []);
+
+  useEffect(() => {
+    getPuzzleInstances();
+  }, [puzzlesList]);
+
+  useEffect(() => {
+    if (!puzzleInstances || !setDisplayPuzzleInstances) return;
+
+    if (!puzzleId || puzzleId === '') {
+      setDisplayPuzzleInstances(puzzleInstances);
+    } else {
```

Comment:
I've been vibing with the use of guard clauses (https://www.youtube.com/watch?v=EumXak7TyQ0) to remove unneeded nesting in functions, I think this would be a good case to utilize it

Reply:
Good one ✨